gbpservice/neutron/nsf/db/nsf_db.py

```diff
@@ -0,0 +1,405 @@
+        try:
+            return self._get_by_id(
+                session, nsf_db_model.NetworkService, network_service_id)
+        except exc.NoResultFound:
+            raise nsf_exc.NetworkServiceNotFound()
+
+    def update_network_service(self, session, network_service_id,
+                               updated_network_service):
+        with session.begin(subtransactions=True):
+            network_service_db = self._get_network_service(
+                session, network_service_id)
+            network_service_db.update(updated_network_service)
+        return self._make_network_service_dict(network_service_db)
+
+    def delete_network_service(self, session, network_service_id):
+        with session.begin(subtransactions=True):
+            network_service_db = self._get_network_service(
+                session, network_service_id)
+            session.delete(network_service_db)
+
+    def get_network_service(self, session, network_service_id, fields=None):
+        service = self._get_network_service(session, network_service_id)
+        return self._make_network_service_dict(service, fields)
+
+    def get_network_services(self, session, filters=None, fields=None,
+                             sorts=None, limit=None, marker=None,
+                             page_reverse=False):
+        marker_obj = self._get_marker_obj(
+            'network_services', limit, marker)
+        return self._get_collection(session, nsf_db_model.NetworkService,
+                                    self._make_network_service_dict,
+                                    filters=filters, fields=fields,
+                                    sorts=sorts, limit=limit,
+                                    marker_obj=marker_obj,
+                                    page_reverse=page_reverse)
+
+    def _set_port_info_for_nsi(self, session, network_service_instance_db,
+                               network_service_instance):
+        nsi_db = network_service_instance_db
+        port_info = network_service_instance.get('port_info')
+        if not port_info:
+            nsi_db.port_info = []
+            return
+        with session.begin(subtransactions=True):
+            nsi_db.port_info = []
+            for port in port_info:
+                port_info_db = nsf_db_model.PortInfo(
+                    id=port['id'],
+                    port_policy=port['port_policy'],
+                    port_classification=port.get('port_classification'),
+                    port_type=port.get('port_type'))
+                session.add(port_info_db)
+                session.flush()  # Any alternatives for flush ??
+                assoc = nsf_db_model.NSIPortAssociation(
+                    network_service_instance_id=(
+                        network_service_instance_db['id']),
+                    data_port_id=port['id'])
+                nsi_db.port_info.append(assoc)
+            del network_service_instance['port_info']
```

Comment:
port info is set as per the DB relation in the above table and removed from network service instance dictionary